src/formatDate.js

```diff
@@ -8,7 +8,70 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldSeparator = fromFormat[3];
+  const newSeparator = toFormat[3];
+  let oldYearFormat = '';
+  const inputDateParts = date.split(oldSeparator);
+  const newDateParts = [];
+
+  if (fromFormat.includes('YY')) {
+    oldYearFormat = 'YY';
+  } else if (fromFormat.includes('YYYY')) {
+    oldYearFormat = 'YYYY';
+  }
+
+  for (let i = 0; i < 3; i++) {
+    const newFormat = toFormat[i];
+    let value = '';
+
+    /**
+     * Define the value of the given date part.
+     * It remains unchanged if the format is the same.
+     * No need to check if new format is equal to 'YY' or 'YYYY' as this is
+     * done while setting the value for oldYearFormat.
+     *
+     * In other case call the 'convertYear' function to reformat the value.
+     */
+    if (['DD', 'MM'].includes(newFormat) || newFormat === oldYearFormat) {
+      const valueIndex = fromFormat.indexOf(newFormat);
+
+      value = inputDateParts[valueIndex];
+    } else if (['YY', 'YYYY'].includes(newFormat)) {
+      const oldValueIndex = fromFormat.indexOf(oldYearFormat);
+      const oldValue = inputDateParts[oldValueIndex];
+
+      value = convertYear(oldValue, newFormat);
+    }
+
+    newDateParts.push(value);
```

Comment:
The loop iterates with a hardcoded limit of 3 which assumes the date will always have three parts. While this is true for most date formats, it would be safer to loop over the length of the 'toFormat' array minus 1 (to exclude the separator), to avoid potential issues if the format changes in the future.